### PR TITLE
Inlinecompletion update

### PR DESCRIPTION
**Pull request by STetsing**
New features

1. Inference Text is limited to 1000 words
2. Introduced `sol-gpt`, working similarly to `gpt`
3. Removed in-browser model
4. Improved `/// `code generation
5. Reduced inline completion calls

### CODE REVIEW

**Review by Aniket-Engg — COMMENTED** [review]
Looks fine to me. Few suggestions:

- Add `sol-gpt` command sample in terminal default text
- Update text `Load & Activate copilot` as we don't anymore show modal to load
- show new suggestion on a new line in certain cases

File: apps/remix-ide/src/app/plugins/copilot/suggestion-service/copilot-suggestion.ts [hunk 1]
```diff
@@ -8,36 +8,31 @@ const profile = {
...
-  useRemoteService(service: string) {
-    this.remoteService = service
+
+  useconfig(config ){
+    this.config = config
   }
 
   discardRemoteService() {
-    this.remoteService = null
+    this.ready = false
```
Comment (yann300) [inline apps/remix-ide/src/app/plugins/copilot/suggestion-service/copilot-suggestion.ts:35]: I think you can remove `discardRemoteService` if we don't use it anymore.

File: apps/remix-ide/src/app/plugins/copilot/suggestion-service/copilot-suggestion.ts [hunk 2]
```diff
@@ -8,36 +8,31 @@ const profile = {
...
-      this.ready = true
-    })
+    this.ready = true // always ready for service
+    this.config = {}
   }
 
-  useRemoteService(service: string) {
-    this.remoteService = service
+
+  useconfig(config ){
```
Comment (yann300) [inline apps/remix-ide/src/app/plugins/copilot/suggestion-service/copilot-suggestion.ts:30]: are you using `useconfig`? if not better ot remove it.

File: apps/remix-ide/src/app/plugins/copilot/suggestion-service/worker.js [hunk 3]
```diff
@@ -35,21 +34,21 @@ self.addEventListener('message', async (event) => {
   if (cmd === 'init') {
     // Retrieve the code-completion pipeline. When called for the first time,
     // this will load the pipeline and save it for future use.
-    CodeCompletionPipeline.model = model
-    await CodeCompletionPipeline.getInstance(x => {
-      // We also add a progress callback to the pipeline so that we can
-      // track model loading.
-      self.postMessage(x);
-    });
+    // CodeCompletionPipeline.model = model
```
Comment (yann300) [inline apps/remix-ide/src/app/plugins/copilot/suggestion-service/worker.js:37]: I think it's better to remove the entire file.

**Pull request closed without merging**